locust/core.py

```diff
@@ -137,35 +141,70 @@ class TaskSetMeta(type):
     Meta class for the main Locust class. It's used to allow Locust classes to specify task execution 
     ratio using an {task:int} dict, or a [(task0,int), ..., (taskN,int)] list.
     """
-    
+
     def __new__(mcs, classname, bases, classDict):
         new_tasks = []
         for base in bases:
             if hasattr(base, "tasks") and base.tasks:
                 new_tasks += base.tasks
-        
+
         if "tasks" in classDict and classDict["tasks"] is not None:
             tasks = classDict["tasks"]
             if isinstance(tasks, dict):
                 tasks = six.iteritems(tasks)
-            
+
             for task in tasks:
                 if isinstance(task, tuple):
                     task, count = task
-                    for i in xrange(0, count):
-                        new_tasks.append(task)
+                    if isinstance(count, int):
+                        setattr(task, 'locust_task_info', {'weight': count, 'order': 2147483647})
+                        for i in xrange(0, count):
+                            new_tasks.append(task)
+                    else:
+                        if 'weight' not in count:
+                            count['weight'] = 1
+                        if 'order' not in count:
+                            count['order'] = 2147483647
+                        setattr(task, 'locust_task_info', count)
+                        for i in xrange(0, count['weight']):
+                            new_tasks.append(task)
                 else:
                     new_tasks.append(task)
-        
+
         for item in six.itervalues(classDict):
-            if hasattr(item, "locust_task_weight"):
-                for i in xrange(0, item.locust_task_weight):
+            if hasattr(item, "locust_task_info"):
+                for i in xrange(0, item.locust_task_info["weight"]):
                     new_tasks.append(item)
-        
+
+        new_tasks = sorted(
+            new_tasks, cmp=lambda left, right: cmp(left.locust_task_info["order"], right.locust_task_info["order"]))
```

Comment:
the `cmp()` function, and the `cmp` parameter to `sorted()` are no longer available in Python3.  You need to use the `key` parameter instead.